pkg/extensions/sync/service.go

```diff
@@ -447,6 +450,67 @@ func (service *BaseService) SyncRepo(ctx context.Context, repo string) error {
 	return nil
 }
 
+// shouldIncludePlatform determines if a platform should be included in the sync
+// based on the configured platforms
+func (service *BaseService) shouldIncludePlatform(platform *ispec.Platform) bool {
+	// Get platform specifications from the configuration
+	var platformSpecs []string
+
+	// Check if we have platform specifications
+	if len(service.config.Platforms) > 0 {
+		platformSpecs = service.config.Platforms
+	} else {
+		// If no platforms are configured, include all
+		return true
+	}
+
+	// If platform is nil, include it
+	if platform == nil || len(platformSpecs) == 0 {
+		return true
+	}
+
+	for _, spec := range platformSpecs {
+		parts := strings.Split(spec, "/")
+		if len(parts) == 2 {
```

Comment:
1. Do we need to handle the variant here as well?
2. Maybe we should have a platform method for matching against a different platform object (including partial matches in case filtering is done just for a subset of platform attributes of course)?